test/integration/forms/form.js

```diff
@@ -1225,6 +1225,119 @@ context('Patient Action Form', function() {
+        fx.data.id = '1';
+        fx.data.relationships.form.data = { id: '11111' };
+        fx.data.relationships['program-action'] = { data: { id: '11111' } };
+        fx.data.relationships['form-responses'].data = [
+          { id: '1', meta: { created_at: testTs() } },
+        ];
+
+        return fx;
+      })
+      .routeForm(_.identity, '11111')
+      .routeFormDefinition()
+      .routeFormActionFields()
+      .routeFormResponse(fx => {
+        fx.data.storyTime = 'Once upon a time...';
+
+        return fx;
+      })
+      .routeActionActivity()
+      .routePatientByAction()
+      .visit('/patient-action/1/form/11111')
+      .wait('@routeAction')
+      .wait('@routeForm')
+      .wait('@routePatientByAction')
+      .wait('@routeFormDefinition')
+      .wait('@routeFormResponse');
+
+    cy
+      .get('.form__controls')
+      .find('button')
+      .contains('Update')
+      .click()
+      .wait('@routeFormActionFields');
+
+    cy
+      .route({
+        status: 201,
+        method: 'POST',
+        delay: 100,
+        url: '/api/form-responses',
+        response: { data: { id: '12345' } },
+      })
+      .as('routePostResponse');
+
+    cy
+      .get('.form__controls')
+      .find('.js-save-button')
+      .should($el => expect($el.text().trim()).to.equal('Save + Go Back'));
+
+    cy
+      .get('.form__controls')
+      .find('.form__save-dropdown-button')
+      .click();
+
+    cy
+      .get('.picklist')
+      .find('.js-picklist-item')
+      .should('have.length', 2)
+      .first()
+      .click()
+      .then(() => {
+        const storage = JSON.parse(localStorage.getItem('form-state_11111'));
+
+        expect(storage.saveButtonType).to.equal('save');
+      });
+
+    cy
+      .get('.form__controls')
+      .find('.js-save-button')
+      .should($el => expect($el.text().trim()).to.equal('Save'));
+
+    cy
+      .get('.form__controls')
+      .find('.form__save-dropdown-button')
+      .click();
+
+    cy
+      .get('.picklist')
+      .find('.js-picklist-item')
+      .should('have.length', 2)
+      .eq(1)
+      .click()
+      .then(() => {
+        const storage = JSON.parse(localStorage.getItem('form-state_11111'));
+
+        expect(storage.saveButtonType).to.equal('saveAndGoBack');
+      });
+
+    cy
+      .get('.form__controls')
+      .find('.js-save-button')
+      .click();
+
+    cy
+      .wait('@routePostResponse')
+      .its('request.body')
+      .should(({ data }) => {
+        expect(data.relationships.action.data.id).to.equal('1');
+        expect(data.relationships.form.data.id).to.equal('11111');
+        expect(data.attributes.response.data.familyHistory).to.equal('Here is some typing');
+        expect(data.attributes.response.data.storyTime).to.equal('Once upon a time...');
+        expect(data.attributes.response.data.patient.fields.weight).to.equal(192);
+      });
+
+    cy
+      .url()
+      .should('contain', 'worklist/owned-by');
+  });
```

Comment:
Brief overview of what I'm doing in this test:

1. Load form page with a basic form loaded. Preset the local storage value to have a `saveButtonType: 'saveAndGoBack'` key-value.
2. Verify the `Save + Go Back` version of the button is shown initially. Based on what is set in local storage.
3. Use the button droplist to switch to the `Save` button. Verify that `saveButtonType: 'save'` is now set in local storage.
4. Use the droplist to switch back to the `Save + Go Back` button and click it to save the form.
5. Make sure form is saved correctly.
6. Check the page URL to ensure the user is taken back to their previous page. In this case, the default `/worklist/owned-by` page is used.